plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -3835,6 +3809,15 @@ public Optional<ConnectorOutputMetadata> finishRefreshMaterializedView(
         commitUpdateAndTransaction(appendFiles, session, transaction, "refresh materialized view");
         transaction = null;
         fromSnapshotForRefresh = Optional.empty();
+
+        // cleanup old snapshots
+        try {
+            executeExpireSnapshots(icebergTable, session, System.currentTimeMillis());
+        }
+        catch (TrinoException e) {
```

Comment:
I think we should catch `Exception` here. Since the write has been already committed, we should not fail the query at this point due to some exception in expiring snapshots.